rust/ql/test/query-tests/unusedentities/main.rs

```diff
@@ -1,118 +1,119 @@
-	i = if cond() { h } else { 5 };
-	println!("use {}", i);
-
-	_ = 1; // (deliberately unused) [NOT DETECTED]
+    let a: i32; // BAD: unused variable
+    let b: i32;
+    let mut c: i32;
+    let mut d: i32;
+    let mut e: i32;
+    let mut f: i32;
+    let g: i32;
+    let h: i32;
+    let i: i32;
+
+    b = 1; // BAD: unused value [NOT DETECTED]
+
+    c = 1; // BAD: unused value [NOT DETECTED]
+    c = 2;
+    println!("use {}", c);
+    c = 3; // BAD: unused value [NOT DETECTED]
+
+    d = 1;
+    if cond() {
+        d = 2; // BAD: unused value [NOT DETECTED]
+        d = 3;
+    } else {
+    }
+    println!("use {}", d);
+
+    e = 1; // BAD: unused value [NOT DETECTED]
+    if cond() {
+        e = 2;
+    } else {
+        e = 3;
+    }
+    println!("use {}", e);
+
+    f = 1;
+    f += 1;
+    println!("use {}", f);
+    f += 1; // BAD: unused value [NOT DETECTED]
+    f = 1;
+    f += 1; // BAD: unused value [NOT DETECTED]
+
+    g = if cond() { 1 } else { 2 }; // BAD: unused value (x2) [NOT DETECTED]
+    h = if cond() { 3 } else { 4 };
+    i = if cond() { h } else { 5 };
+    println!("use {}", i);
+
+    _ = 1; // (deliberately unused) [NOT DETECTED]
 }
 
 // --- structs ---
 
 #[derive(Debug)]
 struct MyStruct {
-	val: i64
+    val: i64,
 }
 
 impl MyStruct {
-	fn my_get(&mut self) -> i64 {
-		return self.val;
-	}
+    fn my_get(&mut self) -> i64 {
+        return self.val;
+    }
 }
 
 fn structs() {
-	let a = MyStruct {val : 1 }; // BAD: unused value [NOT DETECTED]
-	let b = MyStruct {val : 2 };
-	let c = MyStruct {val : 3 };
-	let mut d : MyStruct; // BAD: unused variable [NOT DETECTED]
-	let mut e : MyStruct;
-	let mut f : MyStruct;
-
-	println!("lets use {:?} and {}", b, c.val);
-
-	e = MyStruct {val : 4 };
-	println!("lets use {}", e.my_get());
-	e.val = 5;
-	println!("lets use {}", e.my_get());
-
-	f = MyStruct {val : 6 }; // BAD: unused value [NOT DETECTED]
-	f.val = 7; // BAD: unused value [NOT DETECTED]
+    let a = MyStruct { val: 1 }; // BAD: unused value [NOT DETECTED]
+    let b = MyStruct { val: 2 };
+    let c = MyStruct { val: 3 };
+    let mut d: MyStruct; // BAD: unused variable
+    let mut e: MyStruct;
+    let mut f: MyStruct;
+
+    println!("lets use {:?} and {}", b, c.val);
+
+    e = MyStruct { val: 4 };
+    println!("lets use {}", e.my_get());
+    e.val = 5;
+    println!("lets use {}", e.my_get());
+
+    f = MyStruct { val: 6 }; // BAD: unused value [NOT DETECTED]
+    f.val = 7; // BAD: unused value [NOT DETECTED]
 }
 
 // --- arrays ---
 
 fn arrays() {
-	let is = [1, 2, 3]; // BAD: unused values (x3) [NOT DETECTED]
-	let js = [1, 2, 3];
-	let ks = [1, 2, 3];
+    let is = [1, 2, 3]; // BAD: unused values (x3) [NOT DETECTED]
+    let js = [1, 2, 3];
+    let ks = [1, 2, 3];
 
-	println!("lets use {:?}", js);
+    println!("lets use {:?}", js);
 
-	for k in ks {
-		println!("lets use {}", k);
-	}
+    for k // BAD: unused variable [SPURIOUS: macros not yet supported]
```

Comment:
This one shouldn't be marked as `BAD`.
```suggestion
    for k // SPURIOUS: unused variable (macros not yet supported)
```